### DISCUSSION TITLE
Native AOT support

### DISCUSSION THREAD

**Discussion by Tuttle26** [aws/aws-dotnet-messaging#109] [General]
Is it possible to use this library with AOT in serverless environments?

**Comment by normj** [comment]
@Tuttle26 We haven't gone through the library an addressed any trim warnings from the library. So if you use the library in AOT there is a chance some required code might get trimmed out unless you force turning off trimming. Microsoft guidance is to avoid any trim warnings and if you have any you need to do extra AOT runtime testing that nothing incorrect happen during the Native AOT compilation with trim warnings.

We do have it on our radar to go through the library and address the AOT trim warnings.

**Reply by irunjic** [reply]
Any updates on this issue? This is our main blocker to turn on NativeAOT.

**Comment by normj** [comment]
Native AOT support has been merged in the branch that targets the SDK V4 version. We need some of the changes in V4 to make it complainant. We currently want to GA this library on the V4 version of the SDK.
